bugsnag-android-core/src/main/java/com/bugsnag/android/Client.java

```diff
@@ -216,16 +211,21 @@ public Unit invoke(String activity, Map<String, ?> metadata) {
         }
 
         // register a receiver for automatic breadcrumbs
-        try {
-            Async.run(new Runnable() {
-                @Override
-                public void run() {
-                    IntentFilter intentFilter = SystemBroadcastReceiver.getIntentFilter();
-                    appContext.registerReceiver(systemBroadcastReceiver, intentFilter);
-                }
-            });
-        } catch (RejectedExecutionException ex) {
-            logger.w("Failed to register for automatic breadcrumb broadcasts", ex);
+        final SystemBroadcastReceiver systemBroadcastReceiver =
```

Comment:
I believe this can be assigned to the field, unless there's something I'm missing?

Reply:
I wanted to avoid setting the field if there was nothing to listen to in case attempting to unregister it on `finalize` causes an error (this doesn't seem to be documented).